### ISSUE TITLE
Skipped includes get evaluated

### ISSUE THREAD

**Issue by nukulaar** [ansible/ansible#20287]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

Include

##### ANSIBLE VERSION

```
ansible 2.2.0.0
```

##### SUMMARY
If you skip an include task, it seems that every task in that file gets evaluated, although they will be skipped anyway.

##### STEPS TO REPRODUCE

main.yml
```yaml
- name: "Include a file (This task will be skipped)"
  include: additional_task.yml
  when: false_variable
```

additional_task.yml
```yaml
- name: "Retrieve something"
  shell: ls -1 /etc/sysconfig/network-scripts/ifcfg-*
  register: network_scripts

- name: "Do something"
  file: path={{ item }}
        state=absent
  with_items: "{{ network_scripts.stdout_lines }}"
  when: item == condition
```

##### EXPECTED RESULTS

I would expted that the task in additional_task.yml were not included and dont even show up

##### ACTUAL RESULTS

```
TASK [Retrieve something] *************************
[0;36m skipping: [hostname][0m

TASK [Do something] *******************************
[0;31mfatal: [hostname]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'stdout_lines'"}[0m
	to retry, use: --limit @/var/lib/awx/projects/xxx/taskname.retry
```

##### WORKAROUND
If I set a default to the variable it works again, but it seems pretty unnecassary to try to evaluate every item if the task is skipped anyway.

```yaml
- name: "Retrieve something"
  shell: ls -1 /etc/sysconfig/network-scripts/ifcfg-*
  register: network_scripts

- name: "Do something"
  file: path={{ item }}
        state=absent
  with_items: "{{ network_scripts.stdout_lines | default (' ') }}"
  when: item == condition
```

**Comment by alikins** [comment]
This sounds like the same issue as https://github.com/ansible/ansible/issues/16222  and should be fixed in pending 2.2.1 release.

**Comment by bcoca** [comment]
the problem is that your include is static, which means the `when` is applied to the included tasks (the include itself is NOT skipped). This when fails as it executes per item as per docs http://docs.ansible.com/ansible/playbooks_conditionals.html#loops-and-conditionals

IF you want the `when` to apply to the include itself, try setting `static: no`.
